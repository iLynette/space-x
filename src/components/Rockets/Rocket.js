import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { toggleReserve } from '../../redux/rockets/rockets';

const Rocket = (props) => {
  const {
    rocketId, rocketName, rocketDescription, rocketImage, rocketReserved,
  } = props;

  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(toggleReserve({ id: rocketId }));
  };

  return (
    <div className="rocket-container" key={rocketId}>
      <div>
        <img src={rocketImage} alt="/" />
      </div>
      <div className="content">
        <h1>{rocketName}</h1>
        <p>
          {
        rocketReserved
          && <span className="badge">Reserved</span>

}
          {rocketDescription}
        </p>
        <div className="reserve-rocket">
          {rocketReserved ? (<button type="button" onClick={handleClick} className="btn cancel">Cancel Reservation</button>) : (<button type="button" className="btn" onClick={handleClick}>Reserve Rocket</button>)}
        </div>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocketId: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketDescription: PropTypes.string.isRequired,
  rocketImage: PropTypes.string.isRequired,
  rocketReserved: PropTypes.bool.isRequired,
};
export default Rocket;
