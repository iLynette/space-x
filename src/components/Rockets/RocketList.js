import React from 'react';

function RocketList() {
  return (
    <div>RocketList</div>
  );
}

export default RocketList;
