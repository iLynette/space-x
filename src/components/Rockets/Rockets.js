const Rockets = () => <div />;
export default Rockets;
