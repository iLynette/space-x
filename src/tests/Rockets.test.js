import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Rockets from '../components/Dragons/Dragons';
import RocketList from '../components/Rockets/RocketList';

describe('Rocket tests', () => {
  it('render test', () => {
    const rocket = {
      id: 1,
      image: 'https://img.icons8.com/external-xnimrodx-blue-xnimrodx/64/000000/external-rocket-online-marketing-xnimrodx-blue-xnimrodx.png',
      name: 'falcon1',
      description: 'description for dragon1',
      reserved: false,
    };

    const tree = renderer.create(
      <Provider store={store}>
        <RocketList
          id={rocket.id}
          image={rocket.image}
          name={rocket.name}
          type={rocket.type}
          description={rocket.description}
          reserved={rocket.reserved}
        />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Test if rockets render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
