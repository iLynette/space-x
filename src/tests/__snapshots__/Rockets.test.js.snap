// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket tests Test if rockets render correctly 1`] = `null`;

exports[`Rocket tests render test 1`] = `
<div
  className="rockets"
>
  <ul
    className="rocket-list"
  />
</div>
`;
